presto-main/src/main/java/io/prestosql/testing/MaterializedResult.java

```diff
@@ -397,7 +397,7 @@ public static MaterializedResult materializeSourceDataStream(ConnectorSession se
         while (!pageSource.isFinished()) {
             Page outputPage = pageSource.getNextPage();
             if (outputPage == null) {
-                break;
+                continue;
```

Comment:
That's a bug fix. Please make it into a separate commit.
Is it possible to test? Are we having similar problem somewhere else?

Reply:
Moved to separate commit. Unsure if the issue was a problem anywhere else before but I suspect not since tests passed before this change and still pass afterwards.